catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchSource.java

```diff
@@ -1043,50 +1054,45 @@ public PointRadius getPointRadius() {
   @Nullable
   protected SpatialSearch createCombinedSpatialSearch(
```

Comment:
I would suggest splitting this method into two, one dealing with point radius searches and the other with geometry searches. As it is, the operations performed on each are distinct and do not appear to depend on each other. This will let you get rid of the arrow if/else anti-pattern and since the current method is only called in one place would be minimally disruptive.

Reply:
Also this class is massive, consider refactoring it.

Reply:
while this is good feedback, these comments should be considered ✏️